field_utils/FieldPointerUtilities.F90

```diff
@@ -151,6 +158,40 @@ subroutine assign_fptr_r8_rank2(x, fp_shape, fptr, rc)
       _RETURN(_SUCCESS)
    end subroutine assign_fptr_r8_rank2
 
+   subroutine assign_fptr_r4_rank3(x, fptr, rc)
+      type(ESMF_Field), intent(inout) :: x
+      real(kind=ESMF_KIND_R4), pointer, intent(out) :: fptr(:)
+      integer, optional, intent(out) :: rc
+
+      ! local declarations
+      type(c_ptr) :: cptr
+      integer(ESMF_KIND_I8), allocatable :: fp_shape(:)
+      integer :: status
+
+      fp_shape = get_array_shape(x, _RC)
+      call FieldGetCptr(x, cptr, _RC)
+      call c_f_pointer(cptr, fptr, fp_shape)
+
+      _RETURN(_SUCCESS)
+   end subroutine assign_fptr_r4_rank3
+
+   subroutine assign_fptr_r8_rank3(x, fptr, rc)
+      type(ESMF_Field), intent(inout) :: x
+      real(kind=ESMF_KIND_R8), pointer, intent(out) :: fptr(:)
+      integer, optional, intent(out) :: rc
+
+      ! local declarations
+      type(c_ptr) :: cptr
+      integer(ESMF_KIND_I8), allocatable :: fp_shape(:)
+      integer :: status
+
+      fp_shape = get_array_shape(x, _RC)
+      call FieldGetCptr(x, cptr, _RC)
+      call c_f_pointer(cptr, fptr, fp_shape)
+
+      _RETURN(_SUCCESS)
+   end subroutine assign_fptr_r8_rank3
+
```

Comment:
What is rank3 about these procedures?

Reply:
Partial answer is that `get_array_shape()` is size 3.    Might be clearer to avoid deferred shape (allocatable) in this context.    In any event I think these procedures should go.     I'll explain when we talk.

Reply:
I was referring to the shape in `call c_f_pointer`. I see that you were using `rank` to refer to the rank of the `fptr`. These procedures have the same signature as the rank-1 procedures, so they could not be included in the generic `assign_fptr` funtion. The names are placeholders.